### PR TITLE
r/aws_globalaccelerator_accelerator: Correctly update flow log attributes when enabled

### LINKED ISSUE #14185: Updating Global Accelerator flow logs destination gives InvalidArgumentException
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
10:49 $ terraform -v
Terraform v0.12.25
+ provider.aws v2.69.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
+ provider.tls v2.1.1
```

### Affected Resource(s)

* aws_globalaccelerator_accelerator

### Terraform Configuration Files

```hcl
resource "aws_globalaccelerator_accelerator" "public" {
  name            = "${var.env}public"
  ip_address_type = "IPV4"
  enabled         = true

  attributes {
    flow_logs_enabled   = true
    flow_logs_s3_bucket = "mybucket"
    flow_logs_s3_prefix = "global-accelerator/public-flow-logs2/"
  }

  tags = {
    Environment = var.env
    Projects    = "all"
  }

  lifecycle {
    # Destroying the global accelerator will irrevocably release its public IPs
    prevent_destroy = true
  }
}
```

### Debug Output

FYI this dumps nearly my entire state file to my console... the template shouldn't ask for this.

[Relevant log portion](https://gist.github.com/RulerOf/b6aa171a94f7c821192809ac8d16cfa3)

### Expected Behavior

Terraform should update the flow logs config successfully.

### Actual Behavior

Terraform attempts to update the flow log configuration without disabling it first, which is required by the AWS API.

### Steps to Reproduce

1. Create an `aws_globalaccelerator_accelerator` resource with flow logs enabled.
2. Change the `flow_logs_s3_prefix` attribute such that terraform renders the following `plan` output:

```hcl
# module.public_ingress.aws_globalaccelerator_accelerator.public will be updated in-place
~ resource "aws_globalaccelerator_accelerator" "public" {
      dns_name        = "myuniquednsname.awsglobalaccelerator.com"
      enabled         = true
      hosted_zone_id  = "MYZONEID"
      id              = "arn:aws:globalaccelerator::0123456789012:accelerator/98a4c7a2-bcd2-4d2f-bd47-4ddd402d0725"
      ip_address_type = "IPV4"
      ip_sets         = [
          {
              ip_addresses = [
                  "1.2.3.4",
                  "5.6.7.8",
              ]
              ip_family    = "IPv4"
          },
      ]
      name            = devpublic"
      tags            = {
          "Environment" = "dev"
          "Projects"    = "all"
      }

    ~ attributes {
          flow_logs_enabled   = true
          flow_logs_s3_bucket = "dev-logs"
        ~ flow_logs_s3_prefix = "global-accelerator/frontend-flow-logs/" -> "global-accelerator/public-flow-logs/"
      }
  }
```

### Important Factoids

You can simulate the same behavior and work around it with the AWS CLI:

```
11:23 $ aws globalaccelerator update-accelerator-attributes \
>   --accelerator-arn arn:aws:globalaccelerator::012345678901:accelerator/ec374cd8-f543-4d77-b4ab-346fd617947b \
>   --flow-logs-enabled \
>   --flow-logs-s3-bucket myorgbucket-logs \
>   --flow-logs-s3-prefix global-accelerator/public-flow-logs/

An error occurred (InvalidArgumentException) when calling the UpdateAcceleratorAttributes operation: Subscription already exists: BucketName[myorgbucket-logs] Prefix[global-accelerator/public-flow-logs/]. Please delete and recreate the subscription if you want to change the config of your subscription.
11:23 $ aws globalaccelerator update-accelerator-attributes \
>   --accelerator-arn arn:aws:globalaccelerator::012345678901:accelerator/ec374cd8-f543-4d77-b4ab-346fd617947b \
>   --no-flow-logs-enabled
{
    "AcceleratorAttributes": {
        "FlowLogsEnabled": false,
        "FlowLogsS3Bucket": "myorgbucket-logs",
        "FlowLogsS3Prefix": "global-accelerator/public-flow-logs/"
    }
}
11:23 $ aws globalaccelerator update-accelerator-attributes \
>   --accelerator-arn arn:aws:globalaccelerator::012345678901:accelerator/ec374cd8-f543-4d77-b4ab-346fd617947b \
>   --flow-logs-enabled \
>   --flow-logs-s3-bucket myorgbucket-logs \
>   --flow-logs-s3-prefix global-accelerator/public-flow-logs/
{
    "AcceleratorAttributes": {
        "FlowLogsEnabled": true,
        "FlowLogsS3Bucket": "myorgbucket-logs",
        "FlowLogsS3Prefix": "global-accelerator/public-flow-logs/"
    }
}
```

The relevant code that performs this operation is [here](https://github.com/terraform-providers/terraform-provider-aws/blob/18cc5b509b68565f2309a1171c6da24c371957cc/aws/resource_aws_globalaccelerator_accelerator.go#L327-L349).

It may be possible that adding the following to the top of the function could fix this issue:

```go
opts := &globalaccelerator.UpdateAcceleratorAttributesInput{
  AcceleratorArn:  aws.String(acceleratorArn),
  FlowLogsEnabled: aws.Bool(false),
}

_, err := conn.UpdateAcceleratorAttributes(opts)
if err != nil {
  return fmt.Errorf("Error disabling Global Accelerator flow logs during update operation: %s", err)
}
```

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14185.

Output from acceptance testing:

```console
$ ACCTEST_PARALLELISM=2 make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsGlobalAcceleratorEndpointGroup_\|TestAccAwsGlobalAcceleratorListener_\|TestAccAwsGlobalAcceleratorAccelerator_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAwsGlobalAcceleratorEndpointGroup_\|TestAccAwsGlobalAcceleratorListener_\|TestAccAwsGlobalAcceleratorAccelerator_ -timeout 120m
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_basic
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_basic
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_disappears
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_disappears
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_update
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_update
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_attributes
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_attributes
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_tags
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_tags
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_basic
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_basic
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_disappears
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_disappears
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_Update
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_Update
=== RUN   TestAccAwsGlobalAcceleratorListener_basic
=== PAUSE TestAccAwsGlobalAcceleratorListener_basic
=== RUN   TestAccAwsGlobalAcceleratorListener_disappears
=== PAUSE TestAccAwsGlobalAcceleratorListener_disappears
=== RUN   TestAccAwsGlobalAcceleratorListener_update
=== PAUSE TestAccAwsGlobalAcceleratorListener_update
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_basic
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_basic (71.14s)
=== CONT  TestAccAwsGlobalAcceleratorListener_update
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion (190.26s)
=== CONT  TestAccAwsGlobalAcceleratorListener_disappears
=== CONT  TestAccAwsGlobalAcceleratorListener_basic
--- PASS: TestAccAwsGlobalAcceleratorListener_update (158.63s)
--- PASS: TestAccAwsGlobalAcceleratorListener_disappears (120.03s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_Update
--- PASS: TestAccAwsGlobalAcceleratorListener_basic (184.09s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_Update (257.05s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol (187.01s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_basic
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_basic (174.95s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides (213.72s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint (664.05s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_disappears
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP (708.09s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_attributes
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_disappears (173.58s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_tags
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_tags (91.68s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_update
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_attributes (279.32s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_disappears
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_disappears (115.97s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_update (185.56s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1890.885s
```

### CODE REVIEW

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Changelog comment otherwise LGTM:


messed up output but all tests passed :D
```
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       2636.305s
```

File: .changelog/17739.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/aws_globalaccelerator_accelerator: Allow update of flow log attribute for active flow logs
+```
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply .changelog/17739.txt:3]: Done.

**Review by DrFaust92 — APPROVED** [review]
LGTM!

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @ewbankkit!

Verified Acceptance Tests in Commercial (us-west-2)

```console
ACCTEST_PARALLELISM=2 make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsGlobalAcceleratorEndpointGroup_\|TestAccAwsGlobalAcceleratorListener_\|TestAccAwsGlobalAcceleratorAccelerator_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAwsGlobalAcceleratorEndpointGroup_\|TestAccAwsGlobalAcceleratorListener_\|TestAccAwsGlobalAcceleratorAccelerator_ -timeout 120m
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_basic
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_basic
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_disappears
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_disappears
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_update
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_update
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_attributes
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_attributes
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_tags
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_tags
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_basic
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_basic
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_disappears
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_disappears
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
=== RUN   TestAccAwsGlobalAcceleratorEndpointGroup_Update
=== PAUSE TestAccAwsGlobalAcceleratorEndpointGroup_Update
=== RUN   TestAccAwsGlobalAcceleratorListener_basic
=== PAUSE TestAccAwsGlobalAcceleratorListener_basic
=== RUN   TestAccAwsGlobalAcceleratorListener_disappears
=== PAUSE TestAccAwsGlobalAcceleratorListener_disappears
=== RUN   TestAccAwsGlobalAcceleratorListener_update
=== PAUSE TestAccAwsGlobalAcceleratorListener_update
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_basic
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_basic (67.36s)
=== CONT  TestAccAwsGlobalAcceleratorListener_update
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_MultiRegion (181.51s)
=== CONT  TestAccAwsGlobalAcceleratorListener_disappears
--- PASS: TestAccAwsGlobalAcceleratorListener_update (160.94s)
=== CONT  TestAccAwsGlobalAcceleratorListener_basic
--- PASS: TestAccAwsGlobalAcceleratorListener_disappears (174.83s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_Update
--- PASS: TestAccAwsGlobalAcceleratorListener_basic (164.28s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_Update (211.11s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_TCPHealthCheckProtocol (175.23s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_basic
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_basic (237.50s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_PortOverrides (273.92s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_InstanceEndpoint (553.56s)
=== CONT  TestAccAwsGlobalAcceleratorEndpointGroup_disappears
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_ALBEndpoint_ClientIP (699.12s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_attributes
--- PASS: TestAccAwsGlobalAcceleratorEndpointGroup_disappears (182.40s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_tags
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_tags (87.64s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_update
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_update (165.20s)
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_disappears
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_attributes (292.03s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_disappears (71.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1868.914s
```

**Pull request merged**